napari/components/viewer_model.py

```diff
@@ -152,17 +155,53 @@ def theme(self, theme):
 
     @property
     def grid_size(self):
-        """tuple: Size of grid
-        """
-        return self._grid_size
+        """tuple: Size of grid."""
+        warnings.warn(
+            (
+                "The viewer.grid_size parameter is deprecated and will be removed after version 0.4.2."
```

Comment:
Maybe we start doing deprecations on minor version changes? ie change to 0.5?

Reply:
Agreed 0.4.2 is a little fast and we should leave at least two patch releases (which correspond to at least 4 weeks on our ~2 week release cadence) between every deprecation and removal, and longer for more significant API changes, including a minor release. We should adopt a more formal deprecation policy, maybe @jni you can open an issue capturing your thoughts there